banyan/src/forest/mod.rs

```diff
@@ -244,12 +245,22 @@ impl Config {
         }
     }
 
-    pub fn validate(&self) -> anyhow::Result<()> {
-        anyhow::ensure!(self.max_summary_branches > 1);
-        anyhow::ensure!(self.max_key_branches > 0);
-        anyhow::ensure!(self.target_leaf_size > 0 && self.target_leaf_size <= 1024 * 1024);
-        anyhow::ensure!(self.max_uncompressed_leaf_size <= 16 * 1024 * 1024);
-        anyhow::ensure!(self.zstd_level >= 1 && self.zstd_level <= 22);
+    pub fn validate(&self) -> Result<(), Error> {
+        if !self.max_summary_branches > 1 {
```

Comment:
These expressions read weirdly because the `!` seems so closely attached to the `self.max_summary_branches`. How about `if self.max_summary_branches <= 1`?